lib/events.js

```diff
@@ -1,29 +1,37 @@
 'use strict'
 
-var once = require('once')
+module.exports = function (pinoInstance, internalExtremeHandler) {
+  function theWorldIsBurning (err) {
+    runInternalHandler()
+    handlers.handledOnTerminate = true
+    pinoInstance.onTerminated(this.name, err)
+  }
 
-function onExit (fn) {
-  var oneFn = once(fn)
-  process.on('beforeExit', handle('beforeExit'))
-  process.on('exit', handle('exit'))
-  process.on('uncaughtException', handle('uncaughtException', 1))
-  process.on('SIGHUP', handle('SIGHUP', 129))
-  process.on('SIGINT', handle('SIGINT', 130))
-  process.on('SIGQUIT', handle('SIGQUIT', 131))
-  process.on('SIGTERM', handle('SIGTERM', 143))
-  function handle (evt, code) {
-    onExit.passCode = function (code) {
-      if (oneFn.value) { oneFn = once(fn) }
-      oneFn(code, evt)
-    }
-    onExit.insertCode = function () {
-      if (oneFn.value) { oneFn = once(fn) }
-      oneFn(code, evt)
+  function hup (err) {
+    if (process.listenerCount('SIGHUP') === 1) {
```

Comment:
> @marcbachmann I don't care. Mainline Pino only supports the current LTS releases and up. I'm not concerned with anything other than that.
 
True, there was a discussion about that. Perfectly fine. 👍 
Thereby I'm for the minor change.